src/components/pages/profile/mentorship.js

```diff
@@ -83,15 +83,26 @@ class Mentorship extends React.Component {
         }
     }
 
+	componentWillMount() {
```

Comment:
this should all be done in componentDidMount() i believe. Calling this.setState in componentWillMount doesnt always trigger a rerender. See https://reactjs.org/docs/react-component.html#componentwillmount
Also,
`let newcourses = this.props.mentor.courses.map(obj => obj.name);`
is fine

Reply:
I first had it in componentDidMount but it went into an infinite loop

Reply:
ok nvm i changed it to componentDidMount